includes/flagged-posts.php

```diff
@@ -0,0 +1,338 @@
+ *
+ * Filtering the parent and submenu file doesn't take care of this,
+ * because the Flagged Posts page technically is the Comments page,
+ * presumably.
+ */
+function adminmenu() {
+
+	// Return early if this is not the Flagged Posts page.
+	if ( ! is_flagged_posts_page() ) {
+		return;
+	}
+	?>
+	<script type="text/javascript">
+		function commentsMenuItem() {
+			const item = document.getElementById( 'menu-comments' );
+			const link = item.querySelector( 'a' );
+
+			item.classList.remove( 'current' );
+			link.classList.remove( 'current' );
+			link.removeAttribute( 'aria-current' );
+		};
+
+		commentsMenuItem();
+	</script>
+	<?php
+}
+
+/**
+ * Removes the `unapprove` and `approve` row actions and repurposes `reply`
+ * for use as part of the resolution workflow.
+ *
+ * @param array      $actions Array of comment actions.
+ * @param WP_Comment $comment The comment object.
+ */
+function comment_row_actions( $actions, $comment ) {
+
+	// Return early if this is not the Flagged Posts page.
+	if ( ! is_flagged_posts_page() ) {
+		return $actions;
+	}
+
+	// Return early if the comment is not of the `flagged_content` type.
+	if ( 'flagged_content' !== $comment->comment_type ) {
+		return $actions;
+	}
+
+	// Unset `unapprove`, `approve`, and `spam` actions.
+	unset( $actions['unapprove'] );
+	unset( $actions['approve'] );
+	unset( $actions['spam'] );
+
+	// Modify the default `reply` action to use as a "Resolve" button.
+	$actions['reply'] = sprintf(
+		'<button type="button" onclick="window.commentReply && commentReply.open(\'%s\',\'%s\');" class="vim-r button-link hide-if-no-js" aria-label="%s">%s</button>',
+		$comment->comment_ID,
+		$comment->comment_post_ID,
+		esc_attr__( 'Resolve this flagged content' ),
+		__( 'Resolve' )
+	);
+
+	return $actions;
+}
+
+/**
+ * Add a checkbox to the comment list table reply form for resolving flags.
+ */
+function admin_footer() {
+
+	// Return early if this is not the Flagged Posts page.
+	if ( ! is_flagged_posts_page() ) {
+		return;
+	}
+	?>
+	<script type="text/javascript">
+		function addResolveCheckbox() {
+			const checkbox = document.createElement( 'input' );
+			const label = document.createElement( 'label' );
+			const replyContainer = document.getElementById( 'replycontainer' );
+
+			checkbox.type    = 'checkbox';
+			checkbox.name    = 'resolve_flag';
+			checkbox.id      = 'resolve-flag';
+			checkbox.checked = true;
+
+			label.htmlFor = 'resolve-flag';
+			label.appendChild( document.createTextNode( 'This resolves the concern' ) );
+
+			replyContainer.appendChild( checkbox );
+			replyContainer.appendChild( label );
+		};
+
+		function removeFilterActions() {
+			const filterActions = document.getElementById( 'filter-by-comment-type' ).parentNode;
+
+			filterActions.parentNode.removeChild( filterActions );
+		}
+
+		addResolveCheckbox();
+		removeFilterActions();
+	</script>
+	<?php
+}
+
+/**
+ * If the comment is about bad content, save it as a `flagged_content` type.
```

Comment:
I _think_ this comment needs to be updated to reflect the "resolved" comment type.

Reply:
It does indeed!